components/sup/src/event/nats_message_stream.rs

```diff
@@ -4,93 +4,151 @@ use crate::event::{Error,
+                
+                time::timeout(timeout, self.client.lock().await.connect(supervisor_id, &config))
+                    .await??;
+            } 
+            None => {
+                trace!("Timeout not used");
+                tokio::spawn(async move { self.client.lock().await.connect(supervisor_id, &config).await });
+            }
+        }
+        Ok(())
+    }
+
+    async fn publish(&mut self, subject: &str, msg: impl AsRef<[u8]>) -> io::Result<()> {
+        self.client.lock().await.publish(subject, msg).await
+    }
+}
+
 /// A lightweight handle for the NATS message stream. All events are converted into a NatsMessage
-/// and sent into this stream to be published.
+/// and sentinto this stream to be published.
 ///
 /// An UnboundedSender should be ok here. Messages are continously processed even if the client is
 /// not currently connected.
 pub struct NatsMessageStream(pub(super) UnboundedSender<NatsMessage>);
 
 impl NatsMessageStream {
     pub async fn new(supervisor_id: &str, config: EventStreamConfig) -> Result<NatsMessageStream> {
-        let EventStreamConfig { url,
-                                token,
-                                connect_method,
-                                server_certificate,
-                                .. } = config;
-
-        let mut client = Client::new(vec![url]);
-
-        // Configure the client connect message
-        client.connect_mut()
-              .await
-              .name(format!("hab_client_{}", supervisor_id))
-              .verbose(true)
-              .token(token.to_string());
-
-        // Configure the tls connector
-        let mut tls_connector = TlsConnector::builder();
-        for certificate in habitat_core::tls::native_tls_wrapper::certificates(None)? {
-            tls_connector.add_root_certificate(certificate);
-        }
-        if let Some(certificate) = server_certificate {
-            tls_connector.add_root_certificate(certificate.into());
-        }
-        let tls_connector = tls_connector.build()?;
-        client.set_tls_config(tls_connector).await;
-
         // Connect to the server. If a timeout was set, we want to ensure we establish a connection
         // before exiting the function. If we do not connect within the timeout we return an error.
         // If we do not have a timeout, we dont care if we can immediately connect. Instead we spawn
         // a future that will resolve when a connection is possible. Once we establish a
         // connection, the client will handle reconnecting if necessary.
-        if let Some(timeout) = connect_method.into() {
-            time::timeout(timeout, client.connect()).await
-                                                    .map_err(|_| Error::ConnectNatsServer)?;
-        } else {
-            let client = Client::clone(&client);
-            tokio::spawn(async move { client.connect().await });
-        }
+        let mut client = NatsClient::new();
```

Comment:
With the NatsClient consolidation above, this `client` will not need to be mutable.